### PR TITLE
Adding up budget-not-provided separately #537

### PR DESCRIPTION
**Pull request by Ocre42**
Addressing #537 
Forwardlooking activities with budget-not-provided will be added up separately instead of replacing the activities that provide budgets.

### CODE REVIEW

File: forwardlooking.py [hunk 1]
```diff
@@ -33,34 +33,35 @@ def generate_row(publisher):
...
+                row['year_columns'][1][year] = publisher_stats['forwardlooking_activities_with_budgets'].get(year) or 0
+                if row['budget_not_provided']:
+                    row['year_columns'][1][year] += publisher_stats['forwardlooking_activities_with_budget_not_provided'].get(year) or 0
             else:
                 # Else, use the hierarchy which they are reported at
                 row['year_columns'][0][year] = by_hierarchy[hierarchies_with_nonzero_budgets[0]]['forwardlooking_activities_current'].get(year) or 0
-                row['year_columns'][1][year] = by_hierarchy[hierarchies_with_nonzero_budgets[0]][forwardlooking_budget].get(year) or 0
+                row['year_columns'][1][year] = by_hierarchy[hierarchies_with_nonzero_budgets[0]]['forwardlooking_activities_with_budgets'].get(year) or 0
+                if row['budget_not_provided']:
+                    by_hierarchy[hierarchies_with_nonzero_budgets[0]]['forwardlooking_activities_with_budget_not_provided'].get(year) or 0
```
Comment (andylolz) [inline forwardlooking.py]: Something doesn’t look quite right here – there’s no equals sign on this line. Presumably this is supposed to say:

```python
row['year_columns'][1][year] = …
```

…or similar?

**Comment by andylolz** [comment]
I don’t think this PR attempts to address #537. I.e. it doesn’t attempt to split the figure for activities that include a budget from the figure for activities that explicitly declare `budget-not-provided`.

I think it might instead be addressing https://github.com/IATI/IATI-Dashboard/issues/537#issuecomment-477270154. Is that correct?

**Comment by Ocre42** [comment]
Yes, this addresses the comment specifically, but there is no intention to separate the `budget-not-provided` attribute from the `budgets` in calculations, as this is done to follow the existing methodology.

**Comment by andylolz** [comment]
Okay, that’s fine. I guess #537 can be closed as a wontfix, then.

File: forwardlooking.py (hunk 1 above)
Comment (andylolz) [reply forwardlooking.py]: Looks like this has been addressed in #556.

**Pull request merged**